hierarchy/hierarchy.go

```diff
@@ -33,77 +51,49 @@ func Create(identityToken, stage, parentNodeID, label, description, nodetype, su
 		SubType:     subtype,
 	}
 
-	body, err := json.Marshal(requestBody)
-	if err != nil {
-		err = errors.Wrap(err, "json.Marshal failed")
-		return
-	}
+	req := client.Post("/nodes").
+		WithJSONPayload(requestBody)
 
-	url := fmt.Sprintf(hierarchyBaseURL+"/nodes", stage)
-	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(body))
+	restClient := httpClient(stage, identityToken)
+	resp, err := restClient.Do(ctx, req)
 	if err != nil {
-		err = errors.Wrap(err, "http.NewRequest failed")
-		return
-	}
-
-	req.Header.Set("Authorization", identityToken)
-	req.Header.Set("Content-Type", "application/json")
-
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		err = errors.Wrap(err, "client.Do failed")
-		return
-	}
-
-	defer resp.Body.Close()
-	body, err = ioutil.ReadAll(resp.Body)
-	if err != nil {
-		err = errors.Wrap(err, "ioutil.ReadAll failed")
+		err = errors.Wrap(err, "failed to execute request")
 		return
 	}
 
 	var responseBody struct {
 		ID string `json:"nodeId"`
 	}
-
-	if err = json.Unmarshal(body, &responseBody); err != nil {
-		err = errors.Wrapf(err, "json.Unmarshal failed, body: %s", string(body))
+	if err = resp.Unmarshal(&responseBody); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal response")
 		return
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		err = errors.Errorf("Wrong status: %q, body: %s", resp.Status, string(body))
+		err = errors.Errorf("wrong response status: %q", resp.Status)
 		return
 	}
 
 	return responseBody.ID, nil
 }
 
 func Delete(identityToken, stage, nodeID string) error {
-	url := fmt.Sprintf(hierarchyBaseURL+"/nodes/%s", stage, nodeID)
+	return DeleteWithContext(context.Background(), identityToken, stage, nodeID)
+}
 
-	req, err := http.NewRequest(http.MethodDelete, url, nil)
-	if err != nil {
-		return errors.Wrap(err, "http.NewRequest failed")
-	}
-	req.Header.Set("Authorization", identityToken)
-	req.Header.Set("Content-Type", "application/json")
+func DeleteWithContext(ctx context.Context, identityToken, stage, nodeID string) error {
+	req := client.Delete("/nodes/{id}").
+		Assign("id", nodeID).
+		SetHeader(headers.ContentType, "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	restClient := httpClient(stage, identityToken)
+	resp, err := restClient.Do(ctx, req)
 	if err != nil {
-		return errors.Wrap(err, "client.Do failed")
+		return errors.Wrap(err, "failed to execute request")
```

Comment:
Maybe switch to fmt.Errorf when you are changing stuff

Reply:
Thats would be a different feature. Change all or nothing, not just a subset.